core/components/tinymce/tinymce.class.php

```diff
@@ -228,21 +226,17 @@ public function getScript() {
                 unset($this->properties['elements']);
                 $richtextResource = false; /* workaround for modx ui bug with rte tvs */
             }
-            $this->config['resource'] = $this->config['resource']->toArray();
```

Comment:
This line has to be removed, to fix the MODX Revolution 2.5.1 issue with the current TinyMCE package version.